client/tinychain/ml/mlr.py

```diff
@@ -0,0 +1,95 @@
+
+
+from tinychain.collection.tensor import einsum, Dense , Tensor
+from tinychain.ml import Layer, NeuralNet
+from tinychain.ref import After
+import tinychain as tc 
+import numpy as np
+
+
+class MultipleLinearRegression :
+    
+    @classmethod
+    def load (self,learningrate): 
+        return MultipleLinearRegression([learningrate])
+    
+    @property
+    def learningrate(self):  
+        '''
+        We use property accessor so we recall self.learningrate instead of tc.Number(self[0])
+        '''
+        return tc.Number(self[0])
+    
+    def predict(self, inputs, weights, learning = True):
+        if learning==True:
+            preds = einsum("ij,ki->kj", [weights, inputs]) 
+        else:
+            '''
+            I try to add column of ones ! 
+            '''
+ 
+            inputs_=Tensor.concatenate(1,inputs,axis=1) 
+            preds = einsum("ij,ki->kj", [weights, inputs_])            
+        return preds
+    
+    def cost_func( self, labels ,preds):
+        '''
+        I try to apply Mean Squared Error
+        '''
+        cost = Tensor.sum((labels - preds)**2)/labels.len()
+        return cost
+
+    def gradient_descent(self , inputs , weights, labels):        
+        d_weights = 2/inputs.shape[0]*(einsum("ij,ki->kj", weights, einsum("ij,ki->kj", inputs, inputs.transpose()))-
+                                                      einsum("ij,ki->kj", labels, inputs.transpose()))        
+        return d_weights
+
+
+    def update_params(self ,weights, d_weights): 
+        new_weights = weights - self.learningrate * d_weights
+        return new_weights
+
+    def fit(self, X_train, y_train,iteration):
+        X_train_ = Tensor.concatenate(1,X_train,axis=1) 
+        y_train_ = y_train
+        n_samples, n_features = X_train_.shape[1],X_train_.shape[0]
+        '''
+        Can i create zeros tc tensor  or i have to create it in numpy then convert?
+        '''       
+        weights = np.zeros((n_features + 1, 1)) 
```

Comment:
It's OK to use NumPy in tests but the client should never ever depend on NumPy. You can use `Dense.zeros` here instead if you need.

Reply:
Although I think you should consider using a `DNNLayer` with the `Identity` activation function in order to avoid duplicating logic

Reply:
thats wrong file i uploaded it before in another pull request dunno why its here